crates/api/src/arc.rs

```diff
@@ -0,0 +1,386 @@
+//! Kitsune2 arcs represent a range of hash locations on the DHT.
+//!
+//! An arc can exist in its own right and refer to a range of locations on the DHT.
+//! It can also be used in context, such as an agent. Where it represents the range of locations
+//! that agent is responsible for.
+
+use serde::{Deserialize, Serialize};
+
+/// A basic definition of a storage arc compatible with the concept of
+/// storage and querying of items in a store that fall within that arc.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Default)]
+pub enum BasicArc {
+    /// No DHT locations are contained within this arc.
+    #[default]
+    Empty,
+    /// All DHT locations are contained within this arc.
+    Full,
+    /// A specific range of DHT locations are contained within this arc.
+    ///
+    /// The lower bound is inclusive, the upper bound is exclusive.
```

Comment:
To justify the reasoning for this, which I didn't do on the PR this is split from. Ops should always belong in exactly one arc range when partitioning the hash space. Otherwise you end up with a constant number of single-location overlaps between agents.

I didn't think this through carefully enough when reviewing the original agent definition PR. But I think having closed+open intervals for the arc is the correct logic.

Reply:
You are correct. What's in my head is that I can't set up the arc in an entirely natural way with overlapping ends if they are inclusive bounds. It's not a big adjustment to my code on the other PR to fix that though. I was also thinking that it's nice to have the same boundary logic for time and hash space but time is continuous and the space is not. I'm just going to revert my boundary changes and I'll adjust my partitioning code on the other PR :)